src/main/java/ch/njol/skript/expressions/base/EventValueExpression.java

```diff
@@ -159,24 +121,44 @@ public boolean init() {
 			log.stop();
 		}
 	}
-	
-	@Override
-	public Class<? extends T> getReturnType() {
-		return c;
-	}
-	
+
 	@Override
-	public boolean isSingle() {
-		return single;
+	@Nullable
+	@SuppressWarnings("unchecked")
+	protected T[] get(Event event) {
+		T value = getValue(event);
+		if (value == null)
+			return (T[]) Array.newInstance(c, 0);
+		if (single) {
+			T[] one = (T[]) Array.newInstance(c, 1);
+			one[0] = value;
+			return one;
+		}
+		T[] dataArray = (T[]) value;
+		T[] array = (T[]) Array.newInstance(c.getComponentType(), dataArray.length);
+		System.arraycopy(dataArray, 0, array, 0, array.length);
+		return array;
 	}
-	
-	@Override
-	public String toString(@Nullable Event event, boolean debug) {
-		if (!debug || event == null)
-			return "event-" + Classes.getSuperClassInfo(componentType).getName().toString(!single);
-		return Classes.getDebugMessage(getValue(event));
+
+	@Nullable
+	@SuppressWarnings("unchecked")
+	private <E extends Event> T getValue(E event) {
+		if (converters.containsKey(event.getClass())) {
+			Converter<? super E, ? extends T> converter = (Converter<? super E, ? extends T>) converters.get(event.getClass());
+			return converter == null ? null : converter.convert(event);
```

Comment:
It shouldn't be null if we're checking containsKey right? Would it ever be the case that a value is null?